lib/new_relic/agent/utilization_data.rb

```diff
@@ -84,10 +86,32 @@ def to_collector_hash
         result
       end
 
+      def append_ecs_info(collector_hash)
+        return unless Agent.config[:'utilization.detect_aws']
+
+        Thread.new do
+          # try v4 first, and only try unversioned endpoint if v4 fails
+          ecs = Utilization::ECSV4.new
+          if ecs.detect
+            collector_hash[:vendors] ||= {}
+            collector_hash[:vendors][:ecs] = ecs.metadata
+          else
+            ecs = Utilization::ECS.new
```

Comment:
Is the unversioned endpoint for everything below V4 or above V4 or for something else entirely?

Reply:
V4 is the current one. the unversioned one is only if the V4 one fails for some reason, we'll use unversioned as a backup. It's defined that way in the agent specs, so I didn't look too much futher into the different versions other than that though.

Reply:
Is there an easy way to future-proof this for when a newer version comes out?

Reply:
Beyond the existing logic for the unversioned check we do if v4 fails, I don't think so. It's good that unversioned check is there as a backup in case v4 no longer works, but if something changes with the structure of the data, or the unversioned on changes also, then we don't know what we'd need to support, so we'll have to make changes.